gobblin-core/src/main/java/gobblin/writer/AvroHdfsDataWriter.java

```diff
@@ -170,26 +117,16 @@ public long bytesWritten() throws IOException {
    * Create a new {@link DataFileWriter} for writing Avro records.
    *
    * @param avroFile the Avro file to write to
-   * @param bufferSize the Avro data writer buffer size
    * @param codecFactory a {@link CodecFactory} object for building the compression codec
-   * @param replication the replication factor
-   * @param blockSize the block size
-   * @param permissions a {@link FsPermission} object defining the Avro file permission
    * @throws IOException if there is something wrong creating a new {@link DataFileWriter}
    */
-  private DataFileWriter<GenericRecord> createDatumWriter(Path avroFile, int bufferSize, CodecFactory codecFactory,
-      short replication, long blockSize, FsPermission permissions, Optional<String> group) throws IOException {
-
+  private DataFileWriter<GenericRecord> createDatumWriter(Path avroFile, CodecFactory codecFactory) throws IOException {
     if (this.fs.exists(avroFile)) {
       throw new IOException(String.format("File %s already exists", avroFile));
     }
 
     FSDataOutputStream outputStream =
-        this.fs.create(avroFile, permissions, true, bufferSize, replication, blockSize, null);
-
-    if (group.isPresent()) {
-      this.fs.setOwner(avroFile, this.fs.getFileStatus(avroFile).getOwner(), group.get());
-    }
+        this.fs.create(avroFile, this.permission, true, this.bufferSize, this.replicationFactor, this.blockSize, null);
```

Comment:
Can this be initialized in `FsDataWriter`, and we can make `outputStream` an protected instance variable. This way we can set the group of the file in `FsDataWriter` constructor rather than in the `commit()` method. I think this approach may be cleaner. Also, I think its important to set the group in the constructor because the `commit()` phase won't trigger until all the data is written, so there is a period of time where the file doesn't have the proper group. This could have security implications for the data purger job.